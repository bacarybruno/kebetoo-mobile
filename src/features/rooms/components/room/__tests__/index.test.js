import setupTest from '@app/config/jest-setup'

import Room from '../index'

const givenRoom = setupTest(Room)({
  isOpened: true,
  title: 'Hello World',
  membersCount: 100000,
  message: 'Hello and welcome to kebetoo!',
  caption: '2 days ago',
  room: {
    displayName: 'Kebetoo',
    photoURL: null,
  },
  onPress: jest.fn(),
  theme: 'blue',
})

const givenDiscoverRoom = setupTest(Room.Discover)({
  title: 'Hello World',
  membersCount: 100000,
  message: 'Hello and welcome to kebetoo!',
  caption: '2 days ago',
  room: {
    displayName: 'Kebetoo',
    photoURL: null,
  },
  onPress: jest.fn(),
  theme: 'orange',
  author: 'Jest',
})

it('renders Room', () => {
  const { wrapper } = givenRoom()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('renders discover Room', () => {
  const { wrapper } = givenDiscoverRoom()
  expect(wrapper.toJSON()).toMatchSnapshot()
})
