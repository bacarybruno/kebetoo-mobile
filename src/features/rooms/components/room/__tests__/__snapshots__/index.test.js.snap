// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Room 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "zIndex": 1,
        },
        false,
      ]
    }
  >
    <View
      accessibilityElementsHidden={false}
      importantForAccessibility="auto"
      onLayout={[Function]}
      pointerEvents="box-none"
      style={null}
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "borderBottomColor": "rgb(216, 216, 216)",
              "flex": 1,
              "shadowColor": "rgb(216, 216, 216)",
              "shadowOffset": Object {
                "height": 0.5,
                "width": 0,
              },
              "shadowOpacity": 0.85,
              "shadowRadius": 0,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "maxHeight": undefined,
            "minHeight": undefined,
            "opacity": undefined,
            "transform": undefined,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "height": 0,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "marginHorizontal": 16,
                "opacity": 1,
              }
            }
          >
            <Text
              accessibilityRole="header"
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 17,
                  "fontWeight": "600",
                }
              }
            >
              jest
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
      <View
        accessibilityElementsHidden={false}
        closing={false}
        gestureVelocityImpact={0.3}
        importantForAccessibility="auto"
        onClose={[Function]}
        onGestureBegin={[Function]}
        onGestureCanceled={[Function]}
        onGestureEnd={[Function]}
        onOpen={[Function]}
        onTransitionStart={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        transitionSpec={
          Object {
            "close": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
            "open": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            forwardedRef={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            style={
              Object {
                "flex": 1,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": -1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                  "top": 0,
                  "width": 3,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "borderBottomColor": "rgba(229, 229, 234, 255)",
                          "borderBottomWidth": 0.5,
                          "flexDirection": "row",
                          "paddingHorizontal": 16,
                          "paddingVertical": 16,
                        },
                        false,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "height": 30,
                            "width": 30,
                          },
                          Object {
                            "height": 55,
                            "width": 55,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "rgba(0, 122, 255, 255)",
                              "borderRadius": 1000,
                              "flex": 1,
                              "justifyContent": "center",
                            },
                            undefined,
                            Object {
                              "backgroundColor": "rgba(0, 122, 255, 255)",
                            },
                          ]
                        }
                      >
                        <Text
                          style={
                            Object {
                              "backgroundColor": "transparent",
                              "color": "rgba(255, 255, 255, 255)",
                              "fontFamily": "System",
                              "fontSize": 35,
                              "fontWeight": "600",
                            }
                          }
                        >
                          K
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "justifyContent": "center",
                          "marginLeft": 16,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                          }
                        }
                      >
                        <Text
                          ellipsizeMode="middle"
                          numberOfLines={2}
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "row",
                              "marginBottom": 4,
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontSize": 17,
                                  "fontWeight": "400",
                                  "letterSpacing": -0.408,
                                  "lineHeight": 22,
                                },
                                Array [
                                  Object {
                                    "color": "rgba(0,0,0,0.87)",
                                  },
                                  Object {
                                    "backgroundColor": "transparent",
                                    "fontFamily": "System",
                                    "fontWeight": "600",
                                  },
                                  undefined,
                                ],
                              ]
                            }
                          >
                            Hello World
                          </Text>
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontSize": 15,
                                  "fontWeight": "400",
                                  "letterSpacing": -0.24,
                                  "lineHeight": 20,
                                },
                                Array [
                                  Object {
                                    "color": "rgba(0,0,0,0.87)",
                                  },
                                  undefined,
                                  undefined,
                                ],
                              ]
                            }
                          >
                             • 
                          </Text>
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontSize": 15,
                                  "fontWeight": "400",
                                  "letterSpacing": -0.24,
                                  "lineHeight": 20,
                                },
                                Array [
                                  Object {
                                    "color": "rgba(0,0,0,0.87)",
                                  },
                                  undefined,
                                  undefined,
                                ],
                              ]
                            }
                          >
                            100k members
                          </Text>
                        </Text>
                      </View>
                      <Text
                        numberOfLines={2}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontSize": 15,
                              "fontWeight": "400",
                              "letterSpacing": -0.24,
                              "lineHeight": 20,
                            },
                            Array [
                              Object {
                                "color": "rgba(0,0,0,0.54)",
                              },
                              undefined,
                              undefined,
                            ],
                          ]
                        }
                      >
                        Hello and welcome to kebetoo!
                      </Text>
                      <Text
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontSize": 15,
                              "fontWeight": "400",
                              "letterSpacing": -0.24,
                              "lineHeight": 20,
                            },
                            Array [
                              Object {
                                "color": "rgba(0,0,0,0.38)",
                              },
                              undefined,
                              undefined,
                            ],
                          ]
                        }
                      >
                        2 days ago
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders discover Room 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "zIndex": 1,
        },
        false,
      ]
    }
  >
    <View
      accessibilityElementsHidden={false}
      importantForAccessibility="auto"
      onLayout={[Function]}
      pointerEvents="box-none"
      style={null}
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "borderBottomColor": "rgb(216, 216, 216)",
              "flex": 1,
              "shadowColor": "rgb(216, 216, 216)",
              "shadowOffset": Object {
                "height": 0.5,
                "width": 0,
              },
              "shadowOpacity": 0.85,
              "shadowRadius": 0,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "maxHeight": undefined,
            "minHeight": undefined,
            "opacity": undefined,
            "transform": undefined,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "height": 0,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "marginHorizontal": 16,
                "opacity": 1,
              }
            }
          >
            <Text
              accessibilityRole="header"
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 17,
                  "fontWeight": "600",
                }
              }
            >
              jest
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
      <View
        accessibilityElementsHidden={false}
        closing={false}
        gestureVelocityImpact={0.3}
        importantForAccessibility="auto"
        onClose={[Function]}
        onGestureBegin={[Function]}
        onGestureCanceled={[Function]}
        onGestureEnd={[Function]}
        onOpen={[Function]}
        onTransitionStart={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        transitionSpec={
          Object {
            "close": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
            "open": Object {
              "animation": "spring",
              "config": Object {
                "damping": 500,
                "mass": 3,
                "overshootClamping": true,
                "restDisplacementThreshold": 10,
                "restSpeedThreshold": 10,
                "stiffness": 1000,
              },
            },
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            forwardedRef={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            style={
              Object {
                "flex": 1,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": -1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                  "top": 0,
                  "width": 3,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "borderBottomColor": "rgba(229, 229, 234, 255)",
                        "borderBottomWidth": 0.5,
                        "flexDirection": "row",
                        "paddingHorizontal": 16,
                        "paddingVertical": 16,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "height": 30,
                            "width": 30,
                          },
                          Object {
                            "height": 55,
                            "width": 55,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "rgba(0, 122, 255, 255)",
                              "borderRadius": 1000,
                              "flex": 1,
                              "justifyContent": "center",
                            },
                            undefined,
                            Object {
                              "backgroundColor": "rgba(255, 159, 10, 255)",
                            },
                          ]
                        }
                      >
                        <Text
                          style={
                            Object {
                              "backgroundColor": "transparent",
                              "color": "rgba(255, 255, 255, 255)",
                              "fontFamily": "System",
                              "fontSize": 35,
                              "fontWeight": "600",
                            }
                          }
                        >
                          K
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "justifyContent": "center",
                          "marginLeft": 16,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                          }
                        }
                      >
                        <Text
                          ellipsizeMode="middle"
                          numberOfLines={2}
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "row",
                              "marginBottom": 4,
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontSize": 17,
                                  "fontWeight": "400",
                                  "letterSpacing": -0.408,
                                  "lineHeight": 22,
                                },
                                Array [
                                  Object {
                                    "color": "rgba(0,0,0,0.87)",
                                  },
                                  Object {
                                    "backgroundColor": "transparent",
                                    "fontFamily": "System",
                                    "fontWeight": "600",
                                  },
                                  undefined,
                                ],
                              ]
                            }
                          >
                            Hello World
                          </Text>
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontSize": 15,
                                  "fontWeight": "400",
                                  "letterSpacing": -0.24,
                                  "lineHeight": 20,
                                },
                                Array [
                                  Object {
                                    "color": "rgba(0,0,0,0.87)",
                                  },
                                  undefined,
                                  undefined,
                                ],
                              ]
                            }
                          >
                             • 
                          </Text>
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000000",
                                  "fontFamily": "System",
                                  "fontSize": 15,
                                  "fontWeight": "400",
                                  "letterSpacing": -0.24,
                                  "lineHeight": 20,
                                },
                                Array [
                                  Object {
                                    "color": "rgba(0,0,0,0.87)",
                                  },
                                  undefined,
                                  undefined,
                                ],
                              ]
                            }
                          >
                            100k members
                          </Text>
                        </Text>
                        <Text
                          allowFontScaling={false}
                          style={
                            Object {
                              "color": "rgba(0,0,0,0.87)",
                              "fontFamily": "Ionicons",
                              "fontSize": 25,
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            }
                          }
                          text="ionicon-chevron-forward-outline"
                        />
                      </View>
                      <Text
                        numberOfLines={2}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontSize": 15,
                              "fontWeight": "400",
                              "letterSpacing": -0.24,
                              "lineHeight": 20,
                            },
                            Array [
                              Object {
                                "color": "rgba(0,0,0,0.54)",
                              },
                              undefined,
                              undefined,
                            ],
                          ]
                        }
                      >
                        Created by Jest
                      </Text>
                      <Text
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontSize": 15,
                              "fontWeight": "400",
                              "letterSpacing": -0.24,
                              "lineHeight": 20,
                            },
                            Array [
                              Object {
                                "color": "rgba(0,0,0,0.38)",
                              },
                              undefined,
                              undefined,
                            ],
                          ]
                        }
                      >
                        2 days ago
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
